### PR TITLE
More programming terms

### PR DESCRIPTION
**Pull request by gvwilson**
## Author: 

- @gvwilson 

## Language: 

- English

## Terms defined:

- affordance
- asynchronous
- blob
- block_comment
- build_stale
- callback
- console
- cryptographic_hash_function
- dynamic_loading
- dynamic_lookup
- dynamic_scoping
- exception_handler
- globbing
- lexical_scoping
- license
- lifecycle
- line_comment
- lisp
- log
- log_message
- multi_threaded
- non_blocking_execution
- psf
- python
- r_language
- sandbox
- side_effect
- single_threaded
- stack_overflow
- stream
- synchronous

## Editor

Assign the PR based on the language to the following person:

@zkamvar 

Needs #88, #94, and #95 for completeness.

### CODE REVIEW

File: glossary.yml [hunk 1]
```diff
@@ -358,6 +375,23 @@
...
       [true](#true)/[false](#false). In computing a state of either 0 or 1.
 
+- slug: blob
+  en:
+    term: "binary large object"
+    acronym: "BLOB"
+    def: >
+      Data that is stored in a database without being interpreted in any way,
+      such as an audio file. The term is also now used to refer to data
+      transferred over a network as uninterpreted bits.
```
Comment (zkamvar) [inline glossary.yml:385]: Should there be a mention of this usage in git since it's common to see this term in a github URL?

File: glossary.yml [hunk 2]
```diff
@@ -456,6 +490,14 @@
       A specification for a [build manager](#build_manager) that describes how
       some files depend on others and what to do if those files are out of date.
 
+- slug: build_stale
```
Comment (zkamvar) [inline glossary.yml:493]: Should the slug be `stale_build` since the term is "stale (in build)"? This would follow the convention of [`child_tree`](https://glosario.carpentries.org/en/#child_tree) referring to a "child (in a tree)".

File: glossary.yml [hunk 3]
```diff
@@ -1091,11 +1161,39 @@
...
+      on the [call stack](#call_stack) at the moment the lookup is done. Almost
+      all programming languages use [lexical_scoping](#lexical_scoping) instead,
+      since it is more predictable.
+
 - slug: emacs
   en:
     term: "Emacs (editor)"
     def: >
-      FIXME
+      A [Lisp](#lisp)-based operating system disguised as a text editor that is
```
Comment (zkamvar) [inline glossary.yml:1195]: I'm reading this as a bit of an in-joke among Emacs users that may not necessarily translate for a more general user. Can you rephrase this?

File: glossary.yml [hunk 4]
```diff
@@ -2764,6 +2990,12 @@
       element of the source vector, i.e., `result[index[i]] = source[i]`.  Push
       indexing can easily produce gaps and collisions.
 
+- slug: python
+  en:
+    term: "Python"
+    def: >
+      A popular open-source programming language.
```
Comment (zkamvar) [inline glossary.yml:2997]: This definition felt a bit empty, and I think this may be a good addendum (but needs interpreted and compiled to be defined)

```suggestion
      A popular interpreted open-source programming language that relies on white space indentation to define control structure.
```

File: glossary.yml [hunk 5]
```diff
@@ -3952,9 +4246,9 @@
   en:
     term: "XML"
     def: >
-      A set of rules for defining HTML-like tags and using them to format documents
-      (typically data). XML achieved license plate popularity in the early 2000s,
-      but its complexity led many programmers to adopt [JSON](#json) instead.
+      A set of rules for defining HTML-like tags and using them to format
+      documents (typically data). XML was popular in the early 2000s, but its
```
Comment (zkamvar) [inline glossary.yml:4250]: This answers https://github.com/carpentries/glosario/pull/95#discussion_r497675293, but will unfortunately clash with #95

**Review by zkamvar — DISMISSED** [review]
Overall, good, but I have some questions around slug syntax and definition expansions.

File: glossary.yml (hunk 1 above)
Comment (gvwilson) [reply glossary.yml]: Added - thanks.

File: glossary.yml (hunk 2 above)
Comment (gvwilson) [reply glossary.yml:527]: All of the other terms related to builds have `build_` as a prefix, so I stuck with that. We should probably discuss/standardize - I could see `child_tree` being `tree_child`.

File: glossary.yml (hunk 3 above)
Comment (gvwilson) [reply glossary.yml]: Yeah, sorry about that - fixed.

File: glossary.yml (hunk 4 above)
Comment (gvwilson) [reply glossary.yml]: Added.

File: glossary.yml (hunk 5 above)
Comment (gvwilson) [reply glossary.yml:4506]: I made the same change to #95.

**Review by fmichonneau — APPROVED** [review]
